test-util/src/main/java/io/camunda/zeebe/test/util/bpmn/random/blocks/SubProcessBlockBuilder.java

```diff
@@ -115,10 +115,10 @@ public ExecutionPathSegment generateRandomExecutionPath(final ExecutionPathConte
     final var internalExecutionPath = embeddedSubProcessBuilder.findRandomExecutionPath(context);
 
     if (!hasBoundaryEvents || !internalExecutionPath.canBeInterrupted() || random.nextBoolean()) {
-      result.append(internalExecutionPath);
+      result.append(internalExecutionPath, true);
     } else {
       internalExecutionPath.cutAtRandomPosition(random);
-      result.append(internalExecutionPath);
+      result.append(internalExecutionPath, true);
```

Comment:
I will extract the variable and make it more clear this way. I feel the documentation is not necessary because of the javadoc I've added to the append method.